src/app/wallet/wallet/shared/transaction.service.ts

```diff
@@ -154,7 +168,33 @@ export class TransactionService {
               this.notification.sendNotification(
                 'New stake reward', tx[0]['amount'] + ' PART received');
           }
+          this.checkForNewTransaction(this.txs, tx);
         });
   }
 
+   // Compare old and new transactions to find out updated confirmations
+  compareTransactionResponse(oldTxs: any, newTxs: any) {
+    newTxs.forEach((newtx) => {
+      oldTxs.forEach((oldtx) => {
+        if (oldtx.txid === newtx.txid && oldtx.confirmations !== newtx.confirmations) {
+          oldtx.confirmations = newtx.confirmations;
+        }
+      });
+    });
+  }
+
+  checkForNewTransaction(oldTxs: any, newTxs: any) {
+    const newTransaction = _(newTxs).differenceBy(oldTxs, 'txid').value();
+    newTransaction.map(tx => {
+      this.unShiftTransactions(tx);
+    });
+  }
+
+  unShiftTransactions(json: Object) {
+    if (this.txs.length === 10) {
```

Comment:
Isn't the length variable depending on the paginator?

Reply:
yes 'MAX_TXS_PER_PAGE' initially equals to 10